### ISSUE TITLE
[LOVE] specify the user/group in the exec resource

### ISSUE THREAD

**Issue by purpleidea** [purpleidea/mgmt#221]
#### Welcome:

This is a beginner level patch for someone who is new to golang, and is new to mgmt. It is meant to encourage new programmers to get involved with mgmt. It will probably take you a few hours to a day or so to complete.

#### Intro:

We have an "exec" resource. We'd like to pick the user that commands run as. You'll want to be at least casually familiar with the https://github.com/purpleidea/mgmt/blob/master/docs/resource-guide.md first. Find the exec resource in git master and look at the code.

To implement this, you'll need to either recommend an alternate design, or use the one provided below. This hasn't been exhaustively investigated, so comments are welcome!

#### Design:

It seems the pattern to add this sort of thing might look like:

```
cmd := exec.Command(command, args...)
cmd.SysProcAttr = &syscall.SysProcAttr{}
cmd.SysProcAttr.Credential = &syscall.Credential{Uid: uid, Gid: gid}
```

I haven't tested this, but it looks plausible. You'll need to lookup the uid/gid beforehand. This is currently done in file.go for that code but it might sense to refactor these functions into resources/util.go first and modify to make them generic. This probably relates to https://github.com/purpleidea/mgmt/issues/214 (possible/likely dependency)

This should be fun code and good to orient yourself in a new project and around some new libraries! If you get stuck, ping purpleidea on Freenode IRC.

Read the contributions tips:
https://github.com/purpleidea/mgmt/blob/master/.github/PULL_REQUEST_TEMPLATE.md

Happy hacking, and ping me for questions!

#### Bonus/Future work:

* Add automatic edges ( https://ttboj.wordpress.com/2016/03/14/automatic-edges-in-mgmt/ ) between this resource and the `user` and `group` resources. Group is being tracked in https://github.com/purpleidea/mgmt/issues/220 so as of current writing only user auto edges can be added at this time. Leave any ideas for other automatic edges in the comments.

* Add some tests! It would be nice to have something in resources/exec_test.go and maybe some fancier tests in https://github.com/purpleidea/mgmt/tree/master/test/shell as well.

**Comment by purpleidea** [comment]
Over 80% of this is merged! w00t!

What remains is the automatic edges portion. 1/2 of which is here: https://github.com/purpleidea/mgmt/pull/270/files

**Comment by kwilczynski** [comment]
Hi @purpleidea!

I wanted to take this one, but just noticed that circa two weeks ago it almost got completed. What is outstanding and what else needs to be done?

I am here to help!

**Comment by purpleidea** [comment]
@kwilczynski Hey, great! The autoedges from exec to user/group as shown in PR#270 is only 1/2 done. If you want to send a patch against that, I can merge both. Unless @xiu is finishing it...

Alternatively there are other patches to write :)

**Comment by xiu** [comment]
hey @kwilczynski, you can go ahead and finish it if you want, you will be faster as I'm not really available currently. If you need help don't hesitate to ping me on IRC, I'll answer as soon as possible!

**Comment by arbreezy** [comment]
Hey @purpleidea 
Is this still on ?

**Comment by purpleidea** [comment]
@arbreezy You want to finish the autoedges part??

**Comment by arbreezy** [comment]
Yes @purpleidea, I can have a look early next week. I will ping you on IRC for any questions.

**Comment by purpleidea** [comment]
On Thu, Nov 29, 2018 at 4:52 AM Aris Boutselis <notifications@github.com> wrote:
>
> Yes @purpleidea, I can have a look early next week. I will ping you on IRC for any questions.

Great, sounds good!
